go/vt/wrangler/testlib/init_shard_master_test.go

```diff
@@ -78,6 +78,35 @@ func TestInitMasterShard(t *testing.T) {
 	master.StartActionLoop(t, wr)
 	defer master.StopActionLoop(t)
 
+	// wait for shard record to be updated with master
+	timer := time.NewTimer(10 * time.Second)
+	defer timer.Stop()
+loop:
+	for {
+		select {
+		case <-timer.C:
+			// we timed out
+			si, err := ts.GetShard(ctx, master.Tablet.Keyspace, master.Tablet.Shard)
+			if err != nil {
+				t.Fatalf("GetShard(%v, %v) failed: %v", master.Tablet.Keyspace, master.Tablet.Shard, err)
+			}
+			if !topoproto.TabletAliasEqual(si.MasterAlias, master.Tablet.Alias) {
+				t.Fatalf("ShardInfo should have MasterAlias %v but has %v", topoproto.TabletAliasString(master.Tablet.Alias), topoproto.TabletAliasString(si.MasterAlias))
+			}
+		default:
```

Comment:
Since we aren't also waiting on some other channel operation in addition to the timer, we don't really need a `select`. Because of this `default:` case, the `select` actually never waits; it merely asks, "Is the timer done?" This question could be answered more simply with a loop of the form:

```go
startTime := time.Now()
for {
  if time.Since(startTime) > timeout {
    // fatal timeout error
  }
  if value == wanted {
    break
  }
  time.Sleep(interval)
}
```

Reply:
done